File: testers/avro/java-json/generated-and-checked-in/com/example/events/CustomerId.java
package com.example.events;

import com.fasterxml.jackson.annotation.JsonValue;

/** Customer identifier */
public record CustomerId(@JsonValue Long value) {
  public CustomerId withValue(Long value) {
    return new CustomerId(value);
  }

  @Override
  public java.lang.String toString() {
    return value.toString();
  }

  /** Create a CustomerId from a raw value */
  public static CustomerId valueOf(Long v) {
    return new CustomerId(v);
  }

  /** Get the underlying value */
  public Long unwrap() {
    return this.value();
  }
}

File: testers/avro/scala-json/generated-and-checked-in/com/example/events/CustomerId.scala
package com.example.events

import com.fasterxml.jackson.annotation.JsonValue

/** Customer identifier */
case class CustomerId(@JsonValue value: Long) extends scala.AnyVal {
  /** Get the underlying value */
  def unwrap: Long = {
    return this.value
  }
}

object CustomerId {
  /** Create a CustomerId from a raw value */
  def valueOf(v: Long): CustomerId = {
    return new CustomerId(v)
  }
}